README.md

```diff
@@ -11,10 +11,25 @@ Sample gulpfile.js:
 ```js
 var format = require('gulp-clang-format');
 gulp.task('check-format', function() {
-	return gulp.src('*.js')
+  return gulp.src('*.js')
 	.pipe(format.checkFormat());
 });
 ```
+
+### Promoting warnings to errors
+If you want to enforce the formatting, so that other team members don't introduce
+code that gives you a warning, you can turn them into build errors by acting on
+the 'warning' event. For example, this task exits the build immediately:
+```js
+var format = require('gulp-clang-format');
+gulp.task('check-format', function() {
+  return gulp.src('*.js')
+	.pipe(format.checkFormat('file'))
+    .on('warning', function(e) { process.stdout.write(e.message); process.exit(1) });
```

Comment:
Would it be more convenient for users to call `.pipe(format.checkFormat('file', {failOnFormatError: true})`, and then have the pipeline emit an error? Or is that not possible thanks to Gulp?